fars/booking/views.py

```diff
@@ -196,6 +198,49 @@ def _is_unbookable(self, user, booking):
         return True, ''
 
 
+class TabletView(View):
+    template = 'tablet.html'
+    context = {}
+
+    def dispatch(self, request, bookable):
+        bookable_obj = get_object_or_404(Bookable, id_str=bookable)
+        if not bookable_obj.public and not request.user.is_authenticated:
+            return redirect('{}?next={}'.format(reverse('login'), request.path_info))
+        self.context['bookable'] = bookable_obj
+        self.context['errors'] = 0
+        return super().dispatch(request, bookable)
+
+    def get(self, request, bookable):
+        now = pytz.timezone(TIME_ZONE).localize(datetime.now())
+        booking = Booking()
+        booking.bookable = self.context['bookable']
+        self.context['bookform'] = BookingForm(instance=booking)
+        return render(request, self.template, context=self.context)
+
+
+    def post(self, request, bookable):
+        username = request.POST.get('username')
+        pw = request.POST.get('password')
+        user = authenticate(username=username, password=pw)
+        postdata = request.POST.copy()
+        try:
+            postdata['user'] = user.id
+        except AttributeError:
+            postdata['user'] = -1
+            self.context['errors'] = 1
+            self.context['credential_error'] = 1
+        form = BookingForm(postdata, instance=Booking())
+        if form.is_valid():
+            form.save()
+            return redirect('tablet', bookable=bookable)
+        else:
```

Comment:
```suggestion
        
```

Perhaps removing the `else` and indentation of statements in that scope would make it clearer that they are a part of the end of the function? 
Otherwise you could move the statements under the `if-else` into the `else` clause.